_includes/parse-server/file-adapters.md

```diff
@@ -175,6 +175,94 @@ var api = new ParseServer({
 
 Don't forget to change **S3_ACCESS_KEY**, **S3_SECRET_KEY** and **S3_BUCKET** to their correct value.
 
+##### S3Adapter configuration for Digital Ocean Spaces
+
+Spaces is an S3 equivalent prodivided by Digital Ocean. It's use the same api as S3 so you can use it with the S3 Adapter.
+You just need to change the AWS Endpoint to point to your Spaces endpoint.
+
+```javascript
+...
+var S3Adapter = require('parse-server').S3Adapter;
+var AWS = require("aws-sdk");
+
+//Set Digital Ocean Spaces EndPoint
+const spacesEndpoint = new AWS.Endpoint(process.env.SPACES_ENDPOINT);
+//Define S3 options
+var s3Options = {
+  bucket: process.env.SPACES_BUCKET_NAME,
+  baseUrl: process.env.SPACES_BASE_URL,
+  region: process.env.SPACES_REGION,
+  directAccess: true,
+  globalCacheControl: "public, max-age=31536000",
+  bucketPrefix: process.env.SPACES_BUCKET_PREFIX,
+  s3overrides: {
+    accessKeyId: process.env.SPACES_ACCESS_KEY,
+    secretAccessKey: process.env.SPACES_SECRET_KEY,
```

Comment:
This is obviously bad practice, but without `process.env.` it is at least move ambivalent without suggesting something unsafe.

Reply:
Just asking, Why this is obviously bad practice? What is the alternative option for this? @mtrezza

Reply:
Making secrets accessible via env vars is bad practice because they can be easier compromised.

Reply:
OK, but what would be an alternative way to pass those secrets to my express app code? I can hard-code those into code or read them from the external resources which is something like environment variables. I am just trying to understand the best practice here. @mtrezza

Reply:
I was curious about this and found a [good piece weighing up the different options](https://medium.com/@GaryHarrower/safely-managing-secrets-with-node-js-and-google-secret-manager-1d9a1055b9e4). Not sure what the right think to do here is tho.

Reply:
Well, this is all new for me. Using environment variables for my secrets was the least suspicious thing in my entire setup for years. I think now it is time to digging to learn more. Thanks for the article btw. @TomWFox

Reply:
No problem, new for me too! 🙂